rust/macros/module.rs

```diff
@@ -576,23 +587,36 @@ pub(crate) fn module(ts: TokenStream) -> TokenStream {
                 \"#
             );
 
+            /// # Safety
+            /// This function should only be use as a initcall pointer to called by the kernel at load time.
             #[cfg(not(MODULE))]
             #[doc(hidden)]
             #[no_mangle]
-            pub extern \"C\" fn __{name}_init() -> core::ffi::c_int {{
-                __init()
+            pub unsafe extern \"C\" fn __{name}_init() -> core::ffi::c_int {{
+                // SAFETY: This function will only be called by the kernel at load time.
+                unsafe {{
+                    __init()
+                }}
             }}
 
+            /// # Safety
+            /// This function should only be use as a initcall pointer to called by the kernel when the module gets unloaded.
             #[cfg(not(MODULE))]
             #[doc(hidden)]
             #[no_mangle]
-            pub extern \"C\" fn __{name}_exit() {{
-                __exit()
+            pub unsafe extern \"C\" fn __{name}_exit() {{
+                // SAFETY: This function will only be called by the kernel when the module gets unloaded.
+                unsafe {{
+                    __exit()
+                }}
             }}
 
-            fn __init() -> core::ffi::c_int {{
+            /// # Safety
+            /// This function should only be called by the kernel to ensure exclusive access to the local `static mut __MOD` variable.
```

Comment:
Having said that, this one is not really called by the kernel (in the sense of the C side, like the others above), but from ourselves. Does it need to be clarified a bit?

In addition, normally there is no need to say why the precondition is required -- that is in the implementation (and its safety comments) that uses the precondition. In other words, the "to ensure..." part.

Reply:
Done, explained it a bit more